sc-memory/sc-core/sc_memory_context_manager.c

```diff
@@ -37,102 +46,310 @@ struct _sc_event_emit_params
+  })
+
+/**
+ * @brief Removes access levels for a specific user in the context manager.
+ * @param _user_addr Address of the user.
+ * @param _adding_levels Access levels to be removed.
+ * @return None.
+ */
+#define _sc_context_remove_user_access_levels(_user_addr, _removing_levels) \
+  ({ \
+    sc_monitor_acquire_write(&manager->user_access_levels_monitor); \
+    sc_access_levels _user_levels = \
+        (sc_uint64)sc_hash_table_get(manager->user_access_levels, GINT_TO_POINTER(SC_ADDR_LOCAL_TO_INT(_user_addr))); \
+    _user_levels &= ~(_removing_levels); \
+    sc_hash_table_insert( \
+        manager->user_access_levels, \
+        GINT_TO_POINTER(SC_ADDR_LOCAL_TO_INT(_user_addr)), \
+        GINT_TO_POINTER(_user_levels)); \
+    sc_monitor_release_write(&manager->user_access_levels_monitor); \
+  })
+
+/**
+ * @brief Sets access levels for a specific sc-memory element.
+ * @param _element_addr Address of the sc-memory element.
+ * @param _access_levels Access levels to be set.
+ * @return None.
+ */
+#define _sc_context_set_access_levels_for_element(_element_addr, _access_levels) \
+  ({ \
+    sc_element * _element; \
+    sc_storage_get_element_by_addr(_element_addr, &_element); \
+    if (_element != null_ptr) \
+      _element->flags.access_levels |= _access_levels; \
+  })
 
 /*! Function that creates a memory context for an authenticated user with specified parameters.
- * @param event Pointer to the sc_event triggering the context creation.
- * @param arc_addr sc-address representing created sc-arc to the authenticated user.
+ * @param event Pointer to the sc-event triggering the context creation.
+ * @param initiator_addr sc-address representing user that initiated this request.
+ * @param connector_addr sc-address representing created sc-connector to the user.
+ * @param connector_type sc-type of created sc-connector to the user.
  * @param user_addr sc-address representing the authenticated user for whom the context is created.
  * @returns Returns a result code indicating the success or failure of the operation (SC_RESULT_OK on success).
  * @note This function is called in response to a sc-event and is responsible for creating a new memory context
  * for an authenticated user and establishing a connection between user and the context.
  */
 sc_result _sc_memory_context_manager_on_authentication_request_user(
     sc_event const * event,
-    sc_addr arc_addr,
+    sc_addr initiator_addr,
+    sc_addr connector_addr,
+    sc_type connector_type,
     sc_addr user_addr)
 {
-  sc_unused(&arc_addr);
+  sc_unused(&initiator_addr);
+
+  // Only temporal sc-arcs can be used
+  if (sc_type_check(connector_type, sc_type_arc_pos_const_temp) == SC_FALSE)
+    return SC_RESULT_OK;
+
+  _sc_context_set_access_levels_for_element(connector_addr, SC_CONTEXT_ACCESS_LEVEL_TO_ALL_ACCESS_LEVELS);
+  _sc_context_set_access_levels_for_element(user_addr, SC_CONTEXT_ACCESS_LEVEL_TO_ERASE_ACCESS_LEVELS);
```

Comment:
I couldn’t understand this, it needs to be discussed

Reply:
There are settings of access levels for keynodes that used to set access levels for users. It is needed to check after that we can add arcs from it